Code/Mantid/MantidQt/CustomInterfaces/src/Indirect/ISISDiagnostics.cpp

```diff
@@ -8,498 +8,526 @@
+ *
+ * @param state :: True to show the second range selectors, false to hide
+ */
+void ISISDiagnostics::sliceTwoRanges(QtProperty *, bool state) {
+  m_uiForm.ppRawPlot->getRangeSelector("SliceBackground")->setVisible(state);
+}
+
+/**
+ * Enables/disables the calibration file field and validator
+ *
+ * @param state :: True to enable calibration file, false otherwise
+ */
+void ISISDiagnostics::sliceCalib(bool state) {
+  m_uiForm.dsCalibration->setEnabled(state);
+}
+
+void ISISDiagnostics::rangeSelectorDropped(double min, double max) {
+  MantidWidgets::RangeSelector *from =
+      qobject_cast<MantidWidgets::RangeSelector *>(sender());
 
-  /**
-   * Enables/disables the calibration file field and validator
-   *
-   * @param state :: True to enable calibration file, false otherwise
-   */
-  void ISISDiagnostics::sliceCalib(bool state)
-  {
-    m_uiForm.dsCalibration->setEnabled(state);
+  if (from == m_uiForm.ppRawPlot->getRangeSelector("SlicePeak")) {
+    m_dblManager->setValue(m_properties["PeakStart"], min);
+    m_dblManager->setValue(m_properties["PeakEnd"], max);
+  } else if (from == m_uiForm.ppRawPlot->getRangeSelector("SliceBackground")) {
+    m_dblManager->setValue(m_properties["BackgroundStart"], min);
+    m_dblManager->setValue(m_properties["BackgroundEnd"], max);
   }
+}
 
-  void ISISDiagnostics::rangeSelectorDropped(double min, double max)
-  {
-    MantidWidgets::RangeSelector* from = qobject_cast<MantidWidgets::RangeSelector*>(sender());
-
-    if(from == m_uiForm.ppRawPlot->getRangeSelector("SlicePeak"))
-    {
-      m_dblManager->setValue(m_properties["PeakStart"], min);
-      m_dblManager->setValue(m_properties["PeakEnd"], max);
-    }
-    else if(from == m_uiForm.ppRawPlot->getRangeSelector("SliceBackground"))
-    {
-      m_dblManager->setValue(m_properties["BackgroundStart"], min);
-      m_dblManager->setValue(m_properties["BackgroundEnd"], max);
-    }
+/**
+ * Handles a double property being changed in the property browser.
+ *
+ * @param prop :: Pointer to the QtProperty
+ * @param val :: New value
+ */
+void ISISDiagnostics::doublePropertyChanged(QtProperty *prop, double val) {
+  auto peakRangeSelector = m_uiForm.ppRawPlot->getRangeSelector("SlicePeak");
+  auto backgroundRangeSelector =
+      m_uiForm.ppRawPlot->getRangeSelector("SliceBackground");
+
+  if (prop == m_properties["PeakStart"])
+    peakRangeSelector->setMinimum(val);
+  else if (prop == m_properties["PeakEnd"])
+    peakRangeSelector->setMaximum(val);
+  else if (prop == m_properties["BackgroundStart"])
+    backgroundRangeSelector->setMinimum(val);
+  else if (prop == m_properties["BackgroundEnd"])
+    backgroundRangeSelector->setMaximum(val);
+  else if (prop == m_properties["PreviewSpec"])
+    handleNewFile();
+  else if (prop == m_properties["SpecMin"]) {
+    m_dblManager->setMinimum(m_properties["SpecMax"], val + 1);
+    m_dblManager->setMinimum(m_properties["PreviewSpec"], val + 1);
```

Comment:
Change this to `val` rather than `val + 1` and tell me if that works, if it does I'll commit it.